snappi/snappicommon.py

```diff
@@ -217,6 +339,68 @@ def clone(self):
         """Creates a deep copy of the current object
         """
         return self.__deepcopy__(None)
+    
+    def _validate_required(self):
+        """Validates the required properties are set
+        """
+        if getattr(self, "_REQUIRED", None) is None:
+            return
+        for p in self._REQUIRED:
+            if p in self._properties and self._properties[p] is not None:
+                continue
+            msg = "{} is a mandatory property of {}"\
+                " and should not be set to None".format(
+                p, self.__class__
+            )
+            raise ValueError(msg)
+
+    def _validate_types(self):
+        common_data_types = [list, str, int, float, bool]
+        for property, value in self._properties.items():
+            if property not in self._TYPES:
+                continue
+            details = self._TYPES[property]
+            if value is None and property not in self._DEFAULTS and \
+                property not in self._REQUIRED:
+                continue
+            if 'enum' in details and value not in details['enum']:
+                msg = 'property {} shall be one of these' \
+                    ' {} enum, but got {} at {}'
+                raise TypeError(msg.format(
+                    property, details['enum'], value, self.__class__
+                ))
+            if details['type'] in common_data_types and \
+                'format' not in details:
+                if not isinstance(value, details['type']):
+                    msg = 'property {} shall be of type {},' \
+                        ' but got {} at {}'
+                    raise TypeError(msg.format(
+                        property, details['type'], type(value), self.__class__
+                    ))
+            if details['type'] not in common_data_types:
+                class_name = details['type']
+                module = importlib.import_module(self.__module__)
+                object_class = getattr(module, class_name)
```

Comment:
let's cache it beforehand. python won't import same moduel multiple times, but we shouldn't even try.

Reply:
is there a way to get the class instance, other than this?
as python won't import the same module multiple time, it would imply python uses the cached one right. correct if I am wrong

Reply:
What different modules are we talking about here? To me, it looks we’re only importing snappi so that we can access classes like snappi.FlowDuration.

My suggestion is to not use importlib so many times, since it still spends times searching for module before attempting to import it.